File: home/ubuntu/remitflow/54remitflow-unified-platform/mobile/v97-features/ios/V97Features.swift
import SwiftUI
import Combine

// ─── V97 Models ───────────────────────────────────────────────────────────────

struct VelocityRule: Codable, Identifiable {
    let id: Int
    let name: String
    let maxAmount: Double
    let windowHours: Int
    let action: String // "block" | "flag" | "review"
    let enabled: Bool
}

struct KycLifecycleState: Codable {
    let userId: Int
    let currentStage: String
    let completedAt: TimeInterval?
    let nextAction: String?
    let riskScore: Int
}

struct DocumentRenewal: Codable, Identifiable {
    let id: Int
    let documentType: String
    let expiresAt: TimeInterval
    let status: String
    let daysUntilExpiry: Int
}

struct WebhookDelivery: Codable, Identifiable {
    let id: Int
    let endpointUrl: String
    let eventType: String
    let status: String
    let attempts: Int
    let lastAttemptAt: TimeInterval?
    let nextRetryAt: TimeInterval?
}

struct ApiKeyInfo: Codable, Identifiable {
    let id: Int
    let name: String
    let prefix: String
    let scopes: [String]
    let lastUsedAt: TimeInterval?
    let expiresAt: TimeInterval?
    let isActive: Bool
}

struct BatchPaymentSummary: Codable, Identifiable {
    let id: Int
    let name: String
    let totalItems: Int
    let successCount: Int
    let failedCount: Int
    let pendingCount: Int
    let status: String
    let createdAt: TimeInterval
}

// ─── V97 ViewModel ────────────────────────────────────────────────────────────

@MainActor
class V97ViewModel: ObservableObject {
    @Published var velocityRules: [VelocityRule] = []
    @Published var kycLifecycle: KycLifecycleState?
    @Published var documentRenewals: [DocumentRenewal] = []
    @Published var failedWebhooks: [WebhookDelivery] = []
    @Published var apiKeys: [ApiKeyInfo] = []
    @Published var batchPayments: [BatchPaymentSummary] = []
    @Published var isLoading = false
    @Published var errorMessage: String?

    private let baseURL = "https://remitflow.manus.space"
    private var cancellables = Set<AnyCancellable>()

    init() {
        Task { await loadAll() }
    }

    func loadAll() async {
        isLoading = true
        defer { isLoading = false }
        async let rules = fetchVelocityRules()
        async let lifecycle = fetchKycLifecycle()
        async let renewals = fetchDocumentRenewals()
        async let webhooks = fetchFailedWebhooks()
        async let keys = fetchApiKeys()
        async let batches = fetchBatchPayments()
        velocityRules = (try? await rules) ?? []
        kycLifecycle = try? await lifecycle
        documentRenewals = (try? await renewals) ?? []
        failedWebhooks = (try? await webhooks) ?? []
        apiKeys = (try? await keys) ?? []
        batchPayments = (try? await batches) ?? []
    }

    private func fetchVelocityRules() async throws -> [VelocityRule] {
        try await fetch("/api/trpc/velocityChecks.list")
    }

    private func fetchKycLifecycle() async throws -> KycLifecycleState {
        try await fetch("/api/trpc/kycLifecycle.getMyLifecycle")
    }

    private func fetchDocumentRenewals() async throws -> [DocumentRenewal] {
        try await fetch("/api/trpc/documentVaultRenewal.listMyRenewals")
    }

    private func fetchFailedWebhooks() async throws -> [WebhookDelivery] {
        try await fetch("/api/trpc/webhookRetry.getFailedDeliveries")
    }

    private func fetchApiKeys() async throws -> [ApiKeyInfo] {
        try await fetch("/api/trpc/apiKeyRotation.list")
    }

    private func fetchBatchPayments() async throws -> [BatchPaymentSummary] {
        try await fetch("/api/trpc/batchPaymentV97.list")
    }

    func retryWebhook(deliveryId: Int) async {
        do {
            try await post("/api/trpc/webhookRetry.retryDelivery", body: ["deliveryId": deliveryId])
            failedWebhooks = (try? await fetchFailedWebhooks()) ?? []
        } catch { errorMessage = error.localizedDescription }
    }

    func rotateApiKey(keyId: Int) async {
        do {
            try await post("/api/trpc/apiKeyRotation.rotate", body: ["keyId": keyId])
            apiKeys = (try? await fetchApiKeys()) ?? []
        } catch { errorMessage = error.localizedDescription }
    }

    func retryFailedBatchItems(batchId: Int) async {
        do {
            try await post("/api/trpc/batchPaymentV97.retryFailed", body: ["batchId": batchId])
            batchPayments = (try? await fetchBatchPayments()) ?? []
        } catch { errorMessage = error.localizedDescription }
    }

    private func fetch<T: Decodable>(_ path: String) async throws -> T {
        guard let url = URL(string: baseURL + path) else { throw URLError(.badURL) }
        var request = URLRequest(url: url)
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        let (data, _) = try await URLSession.shared.data(for: request)
        return try JSONDecoder().decode(T.self, from: data)
    }

    private func post(_ path: String, body: [String: Any]) async throws {
        guard let url = URL(string: baseURL + path) else { throw URLError(.badURL) }
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.httpBody = try JSONSerialization.data(withJSONObject: body)
        let (_, response) = try await URLSession.shared.data(for: request)
        guard let httpResponse = response as? HTTPURLResponse, httpResponse.statusCode < 400 else {
            throw URLError(.badServerResponse)
        }
    }
}

// ─── Velocity Check View ──────────────────────────────────────────────────────

struct VelocityCheckView: View {
    @StateObject private var viewModel = V97ViewModel()

    var body: some View {
        NavigationStack {
            List(viewModel.velocityRules) { rule in
                VelocityRuleRow(rule: rule)
            }
            .navigationTitle("Velocity Rules")
            .overlay {
                if viewModel.isLoading { ProgressView() }
            }
            .refreshable { await viewModel.loadAll() }
        }
    }
}

struct VelocityRuleRow: View {
    let rule: VelocityRule

    var actionColor: Color {
        switch rule.action {
        case "block": return .red
        case "flag": return .orange
        default: return .blue
        }
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack {
                Text(rule.name).font(.headline)
                Spacer()
                Text(rule.action.uppercased())
                    .font(.caption.bold())
                    .foregroundColor(.white)
                    .padding(.horizontal, 8)
                    .padding(.vertical, 2)
                    .background(actionColor)
                    .clipShape(Capsule())
            }
            Text("Max: $\(rule.maxAmount, specifier: "%.0f") / \(rule.windowHours)h window")
                .font(.subheadline)
                .foregroundColor(.secondary)
            Text(rule.enabled ? "Active" : "Disabled")
                .font(.caption)
                .foregroundColor(rule.enabled ? .green : .gray)
        }
        .padding(.vertical, 4)
    }
}

// ─── KYC Lifecycle View ───────────────────────────────────────────────────────

struct KycLifecycleView: View {
    @StateObject private var viewModel = V97ViewModel()

    var body: some View {
        NavigationStack {
            ScrollView {
                if let lifecycle = viewModel.kycLifecycle {
                    VStack(spacing: 16) {
                        KycStageCard(lifecycle: lifecycle)
                        KycRiskScoreCard(riskScore: lifecycle.riskScore)
                        if let nextAction = lifecycle.nextAction {
                            KycNextActionCard(action: nextAction)
                        }
                    }
                    .padding()
                } else {
                    ProgressView()
                }
            }
            .navigationTitle("KYC Lifecycle")
            .refreshable { await viewModel.loadAll() }
        }
    }
}

struct KycStageCard: View {
    let lifecycle: KycLifecycleState

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text("Current Stage").font(.caption).foregroundColor(.secondary)
            Text(lifecycle.currentStage.uppercased())
                .font(.title2.bold())
                .foregroundColor(.blue)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding()
        .background(Color(.systemBackground))
        .cornerRadius(12)
        .shadow(color: .black.opacity(0.05), radius: 4)
    }
}

struct KycRiskScoreCard: View {
    let riskScore: Int

    var scoreColor: Color {
        switch riskScore {
        case 0..<30: return .green
        case 30..<70: return .orange
        default: return .red
        }
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            Text("Risk Score").font(.caption).foregroundColor(.secondary)
            HStack {
                Text("\(riskScore)").font(.title.bold()).foregroundColor(scoreColor)
                Text("/ 100").font(.title3).foregroundColor(.secondary)
                Spacer()
                Gauge(value: Double(riskScore), in: 0...100) {}
                    .gaugeStyle(.accessoryCircularCapacity)
                    .tint(scoreColor)
                    .frame(width: 50, height: 50)
            }
        }
        .padding()
        .background(Color(.systemBackground))
        .cornerRadius(12)
        .shadow(color: .black.opacity(0.05), radius: 4)
    }
}

struct KycNextActionCard: View {
    let action: String

    var body: some View {
        HStack {
            Image(systemName: "arrow.right.circle.fill")
                .foregroundColor(.blue)
            Text(action).font(.subheadline)
            Spacer()
        }
        .padding()
        .background(Color.blue.opacity(0.1))
        .cornerRadius(12)
    }
}

// ─── Document Renewal View ────────────────────────────────────────────────────

struct DocumentRenewalView: View {
    @StateObject private var viewModel = V97ViewModel()

    var body: some View {
        NavigationStack {
            List(viewModel.documentRenewals) { renewal in
                DocumentRenewalRow(renewal: renewal)
            }
            .navigationTitle("Document Renewals")
            .refreshable { await viewModel.loadAll() }
        }
    }
}

struct DocumentRenewalRow: View {
    let renewal: DocumentRenewal

    var urgencyColor: Color {
        switch renewal.daysUntilExpiry {
        case ..<7: return .red
        case ..<30: return .orange
        default: return .green
        }
    }

    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: 4) {
                Text(renewal.documentType).font(.headline)
                Text("\(renewal.daysUntilExpiry) days until expiry")
                    .font(.subheadline)
                    .foregroundColor(urgencyColor)
            }
            Spacer()
            Text(renewal.status.uppercased())
                .font(.caption.bold())
                .foregroundColor(.white)
                .padding(.horizontal, 8)
                .padding(.vertical, 2)
                .background(urgencyColor)
                .clipShape(Capsule())
        }
        .padding(.vertical, 4)
    }
}

// ─── Webhook Retry View ───────────────────────────────────────────────────────

struct WebhookRetryView: View {
    @StateObject private var viewModel = V97ViewModel()

    var body: some View {
        NavigationStack {
            List(viewModel.failedWebhooks) { delivery in
                WebhookDeliveryRow(delivery: delivery) {
                    Task { await viewModel.retryWebhook(deliveryId: delivery.id) }
                }
            }
            .navigationTitle("Failed Webhooks")
            .refreshable { await viewModel.loadAll() }
        }
    }
}

struct WebhookDeliveryRow: View {
    let delivery: WebhookDelivery
    let onRetry: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(delivery.eventType).font(.headline)
            Text(delivery.endpointUrl)
                .font(.caption)
                .foregroundColor(.secondary)
                .lineLimit(1)
            HStack {
                Text("Attempts: \(delivery.attempts)").font(.subheadline)
                Spacer()
                Button("Retry", action: onRetry)
                    .buttonStyle(.borderedProminent)
                    .controlSize(.small)
            }
        }
        .padding(.vertical, 4)
    }
}

// ─── API Key Management View ──────────────────────────────────────────────────

struct ApiKeyManagementView: View {
    @StateObject private var viewModel = V97ViewModel()
    @State private var keyToRotate: ApiKeyInfo?

    var body: some View {
        NavigationStack {
            List(viewModel.apiKeys) { key in
                ApiKeyRow(key: key) { keyToRotate = key }
            }
            .navigationTitle("API Keys")
            .refreshable { await viewModel.loadAll() }
            .confirmationDialog(
                "Rotate API Key",
                isPresented: Binding(
                    get: { keyToRotate != nil },
                    set: { if !$0 { keyToRotate = nil } }
                ),
                titleVisibility: .visible
            ) {
                Button("Rotate Key", role: .destructive) {
                    if let key = keyToRotate {
                        Task { await viewModel.rotateApiKey(keyId: key.id) }
                    }
                    keyToRotate = nil
                }
                Button("Cancel", role: .cancel) { keyToRotate = nil }
            } message: {
                Text("This will invalidate the current key and generate a new one.")
            }
        }
    }
}

struct ApiKeyRow: View {
    let key: ApiKeyInfo
    let onRotate: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack {
                Text(key.name).font(.headline)
                Spacer()
                Button("Rotate", action: onRotate)
                    .buttonStyle(.bordered)
                    .controlSize(.small)
            }
            Text("\(key.prefix)••••••••")
                .font(.system(.subheadline, design: .monospaced))
                .foregroundColor(.secondary)
            Text("Scopes: \(key.scopes.joined(separator: ", "))")
                .font(.caption)
                .foregroundColor(.secondary)
        }
        .padding(.vertical, 4)
    }
}

// ─── V97 Features Hub ─────────────────────────────────────────────────────────

struct V97FeaturesHub: View {
    var body: some View {
        NavigationStack {
            List {
                Section("Compliance & Risk") {
                    NavigationLink("Velocity Rules", destination: VelocityCheckView())
                    NavigationLink("KYC Lifecycle", destination: KycLifecycleView())
                }
                Section("Documents") {
                    NavigationLink("Document Renewals", destination: DocumentRenewalView())
                }
                Section("Integrations") {
                    NavigationLink("Webhook Retry", destination: WebhookRetryView())
                    NavigationLink("API Keys", destination: ApiKeyManagementView())
                }
            }
            .navigationTitle("Advanced Features")
        }
    }
}

#Preview {
    V97FeaturesHub()
}

File: 54remitflow-unified-platform/mobile/v97-features/android/V97Features.kt
package com.remitflow.mobile.v97

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch
import retrofit2.http.*

// ─── V97 API Models ───────────────────────────────────────────────────────────

data class VelocityRule(
    val id: Int,
    val name: String,
    val maxAmount: Double,
    val windowHours: Int,
    val action: String, // "block" | "flag" | "review"
    val enabled: Boolean
)

data class KycLifecycleState(
    val userId: Int,
    val currentStage: String,
    val completedAt: Long?,
    val nextAction: String?,
    val riskScore: Int
)

data class DocumentRenewal(
    val id: Int,
    val documentType: String,
    val expiresAt: Long,
    val status: String, // "pending" | "completed" | "overdue"
    val daysUntilExpiry: Int
)

data class WebhookDelivery(
    val id: Int,
    val endpointUrl: String,
    val eventType: String,
    val status: String,
    val attempts: Int,
    val lastAttemptAt: Long?,
    val nextRetryAt: Long?
)

data class ApiKeyInfo(
    val id: Int,
    val name: String,
    val prefix: String,
    val scopes: List<String>,
    val lastUsedAt: Long?,
    val expiresAt: Long?,
    val isActive: Boolean
)

data class BatchPaymentSummary(
    val id: Int,
    val name: String,
    val totalItems: Int,
    val successCount: Int,
    val failedCount: Int,
    val pendingCount: Int,
    val status: String,
    val createdAt: Long
)

// ─── V97 API Interface ────────────────────────────────────────────────────────

interface V97ApiService {
    @GET("api/trpc/velocityChecks.list")
    suspend fun getVelocityRules(): List<VelocityRule>

    @GET("api/trpc/kycLifecycle.getMyLifecycle")
    suspend fun getKycLifecycle(): KycLifecycleState

    @GET("api/trpc/documentVaultRenewal.listMyRenewals")
    suspend fun getDocumentRenewals(): List<DocumentRenewal>

    @GET("api/trpc/webhookRetry.getFailedDeliveries")
    suspend fun getFailedWebhooks(): List<WebhookDelivery>

    @POST("api/trpc/webhookRetry.retryDelivery")
    suspend fun retryWebhook(@Body body: Map<String, Int>): Map<String, Any>

    @GET("api/trpc/apiKeyRotation.list")
    suspend fun getApiKeys(): List<ApiKeyInfo>

    @POST("api/trpc/apiKeyRotation.rotate")
    suspend fun rotateApiKey(@Body body: Map<String, Int>): Map<String, String>

    @GET("api/trpc/batchPaymentV97.list")
    suspend fun getBatchPayments(): List<BatchPaymentSummary>

    @POST("api/trpc/batchPaymentV97.retryFailed")
    suspend fun retryFailedBatchItems(@Body body: Map<String, Int>): Map<String, Any>
}

// ─── Velocity Check Screen ────────────────────────────────────────────────────

@Composable
fun VelocityCheckScreen(viewModel: V97ViewModel) {
    val rules by viewModel.velocityRules.collectAsState()
    val isLoading by viewModel.isLoading.collectAsState()

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text("Velocity Rules", style = MaterialTheme.typography.headlineMedium)
        Spacer(modifier = Modifier.height(8.dp))

        if (isLoading) {
            CircularProgressIndicator(modifier = Modifier.align(Alignment.CenterHorizontally))
        } else {
            LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                items(rules) { rule ->
                    VelocityRuleCard(rule = rule)
                }
            }
        }
    }
}

@Composable
fun VelocityRuleCard(rule: VelocityRule) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(rule.name, style = MaterialTheme.typography.titleMedium)
                Badge(
                    containerColor = when (rule.action) {
                        "block" -> Color(0xFFEF4444)
                        "flag" -> Color(0xFFF59E0B)
                        else -> Color(0xFF3B82F6)
                    }
                ) {
                    Text(rule.action.uppercase(), color = Color.White)
                }
            }
            Spacer(modifier = Modifier.height(4.dp))
            Text("Max: \$${rule.maxAmount} / ${rule.windowHours}h window",
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant)
            Text("Status: ${if (rule.enabled) "Active" else "Disabled"}",
                style = MaterialTheme.typography.bodySmall)
        }
    }
}

// ─── KYC Lifecycle Screen ─────────────────────────────────────────────────────

@Composable
fun KycLifecycleScreen(viewModel: V97ViewModel) {
    val lifecycle by viewModel.kycLifecycle.collectAsState()

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text("KYC Lifecycle", style = MaterialTheme.typography.headlineMedium)
        Spacer(modifier = Modifier.height(16.dp))

        lifecycle?.let { state ->
            Card(modifier = Modifier.fillMaxWidth()) {
                Column(modifier = Modifier.padding(16.dp)) {
                    Text("Current Stage", style = MaterialTheme.typography.labelMedium)
                    Text(state.currentStage.uppercase(),
                        style = MaterialTheme.typography.headlineSmall,
                        color = MaterialTheme.colorScheme.primary)
                    Spacer(modifier = Modifier.height(8.dp))
                    Text("Risk Score: ${state.riskScore}/100",
                        style = MaterialTheme.typography.bodyMedium)
                    state.nextAction?.let { action ->
                        Spacer(modifier = Modifier.height(8.dp))
                        Text("Next Action: $action",
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.secondary)
                    }
                }
            }
        } ?: CircularProgressIndicator()
    }
}

// ─── Document Renewal Screen ──────────────────────────────────────────────────

@Composable
fun DocumentRenewalScreen(viewModel: V97ViewModel) {
    val renewals by viewModel.documentRenewals.collectAsState()

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text("Document Renewals", style = MaterialTheme.typography.headlineMedium)
        Spacer(modifier = Modifier.height(8.dp))

        LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            items(renewals) { renewal ->
                DocumentRenewalCard(renewal = renewal)
            }
        }
    }
}

@Composable
fun DocumentRenewalCard(renewal: DocumentRenewal) {
    val urgencyColor = when {
        renewal.daysUntilExpiry <= 7 -> Color(0xFFEF4444)
        renewal.daysUntilExpiry <= 30 -> Color(0xFFF59E0B)
        else -> Color(0xFF10B981)
    }

    Card(modifier = Modifier.fillMaxWidth()) {
        Row(
            modifier = Modifier.padding(16.dp).fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column {
                Text(renewal.documentType, style = MaterialTheme.typography.titleMedium)
                Text("${renewal.daysUntilExpiry} days until expiry",
                    style = MaterialTheme.typography.bodySmall,
                    color = urgencyColor)
            }
            Badge(containerColor = urgencyColor) {
                Text(renewal.status.uppercase(), color = Color.White)
            }
        }
    }
}

// ─── Webhook Retry Screen ─────────────────────────────────────────────────────

@Composable
fun WebhookRetryScreen(viewModel: V97ViewModel) {
    val deliveries by viewModel.failedWebhooks.collectAsState()

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text("Failed Webhooks", style = MaterialTheme.typography.headlineMedium)
        Spacer(modifier = Modifier.height(8.dp))

        LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            items(deliveries) { delivery ->
                WebhookDeliveryCard(
                    delivery = delivery,
                    onRetry = { viewModel.retryWebhook(delivery.id) }
                )
            }
        }
    }
}

@Composable
fun WebhookDeliveryCard(delivery: WebhookDelivery, onRetry: () -> Unit) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(16.dp)) {
            Text(delivery.eventType, style = MaterialTheme.typography.titleMedium)
            Text(delivery.endpointUrl,
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant)
            Spacer(modifier = Modifier.height(4.dp))
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text("Attempts: ${delivery.attempts}", style = MaterialTheme.typography.bodySmall)
                Button(onClick = onRetry, modifier = Modifier.height(32.dp)) {
                    Text("Retry")
                }
            }
        }
    }
}

// ─── API Key Management Screen ────────────────────────────────────────────────

@Composable
fun ApiKeyManagementScreen(viewModel: V97ViewModel) {
    val apiKeys by viewModel.apiKeys.collectAsState()
    var showRotateDialog by remember { mutableStateOf<Int?>(null) }

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text("API Keys", style = MaterialTheme.typography.headlineMedium)
        Spacer(modifier = Modifier.height(8.dp))

        LazyColumn(verticalArrangement = Arrangement.spacedBy(8.dp)) {
            items(apiKeys) { key ->
                ApiKeyCard(
                    apiKey = key,
                    onRotate = { showRotateDialog = key.id }
                )
            }
        }
    }

    showRotateDialog?.let { keyId ->
        AlertDialog(
            onDismissRequest = { showRotateDialog = null },
            title = { Text("Rotate API Key") },
            text = { Text("This will invalidate the current key and generate a new one. Are you sure?") },
            confirmButton = {
                TextButton(onClick = {
                    viewModel.rotateApiKey(keyId)
                    showRotateDialog = null
                }) { Text("Rotate") }
            },
            dismissButton = {
                TextButton(onClick = { showRotateDialog = null }) { Text("Cancel") }
            }
        )
    }
}

@Composable
fun ApiKeyCard(apiKey: ApiKeyInfo, onRotate: () -> Unit) {
    Card(modifier = Modifier.fillMaxWidth()) {
        Column(modifier = Modifier.padding(16.dp)) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Column {
                    Text(apiKey.name, style = MaterialTheme.typography.titleMedium)
                    Text("${apiKey.prefix}••••••••",
                        style = MaterialTheme.typography.bodyMedium,
                        color = MaterialTheme.colorScheme.onSurfaceVariant)
                }
                OutlinedButton(onClick = onRotate) {
                    Text("Rotate")
                }
            }
            Spacer(modifier = Modifier.height(4.dp))
            Text("Scopes: ${apiKey.scopes.joinToString(", ")}",
                style = MaterialTheme.typography.bodySmall)
        }
    }
}

// ─── V97 ViewModel ────────────────────────────────────────────────────────────

class V97ViewModel(private val api: V97ApiService) : ViewModel() {
    private val _velocityRules = MutableStateFlow<List<VelocityRule>>(emptyList())
    val velocityRules: StateFlow<List<VelocityRule>> = _velocityRules

    private val _kycLifecycle = MutableStateFlow<KycLifecycleState?>(null)
    val kycLifecycle: StateFlow<KycLifecycleState?> = _kycLifecycle

    private val _documentRenewals = MutableStateFlow<List<DocumentRenewal>>(emptyList())
    val documentRenewals: StateFlow<List<DocumentRenewal>> = _documentRenewals

    private val _failedWebhooks = MutableStateFlow<List<WebhookDelivery>>(emptyList())
    val failedWebhooks: StateFlow<List<WebhookDelivery>> = _failedWebhooks

    private val _apiKeys = MutableStateFlow<List<ApiKeyInfo>>(emptyList())
    val apiKeys: StateFlow<List<ApiKeyInfo>> = _apiKeys

    private val _isLoading = MutableStateFlow(false)
    val isLoading: StateFlow<Boolean> = _isLoading

    init {
        loadAll()
    }

    private fun loadAll() {
        viewModelScope.launch {
            _isLoading.value = true
            try {
                _velocityRules.value = api.getVelocityRules()
                _kycLifecycle.value = api.getKycLifecycle()
                _documentRenewals.value = api.getDocumentRenewals()
                _failedWebhooks.value = api.getFailedWebhooks()
                _apiKeys.value = api.getApiKeys()
            } catch (e: Exception) {
                // Handle error
            } finally {
                _isLoading.value = false
            }
        }
    }

    fun retryWebhook(deliveryId: Int) {
        viewModelScope.launch {
            try {
                api.retryWebhook(mapOf("deliveryId" to deliveryId))
                _failedWebhooks.value = api.getFailedWebhooks()
            } catch (e: Exception) { /* handle */ }
        }
    }

    fun rotateApiKey(keyId: Int) {
        viewModelScope.launch {
            try {
                api.rotateApiKey(mapOf("keyId" to keyId))
                _apiKeys.value = api.getApiKeys()
            } catch (e: Exception) { /* handle */ }
        }
    }
}
